### PR TITLE
mds: Reorganize class members in DamageTable header

### PR DESCRIPTION
**Pull request by varshar16**
Fixes: https://tracker.ceph.com/issues/41178
Signed-off-by: Varsha Rao <varao@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/mds/DamageTable.h [hunk 1]
```diff
@@ -33,71 +31,70 @@ typedef enum
 
 } damage_entry_type_t;
 
+class CDir;
+
```
Comment (joscollin) [inline src/mds/DamageTable.h]: I don't see a need for a forward declaration here. CDir is defined in another file. As you are touching it, replace it with a `#include` ?

Comment (varshar16) [reply src/mds/DamageTable.h]: I have replaced it.

**Comment by joscollin** [comment]
jenkins test make check

File: src/mds/DamageTable.h [hunk 2]
```diff
@@ -20,8 +20,7 @@
 
 #include "mdstypes.h"
 #include "include/random.h"
-
-class CDir;
+#include "CDir.h"
```
Comment (batrick) [inline src/mds/DamageTable.h]: Why was this change necessary?

Comment (joscollin) [reply src/mds/DamageTable.h]: oops! My bad. That was just a pointer. Please revert it.
